metriq_gym/run.py

```diff
@@ -11,29 +13,75 @@
 logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s")
 
 
-def main():
+def main() -> int:
+    """Main entry point for the CLI."""
     load_dotenv()
     args = parse_arguments()
-    job_manager = JobManager(args.jobs_file)
+    job_manager = JobManager()
 
     if args.action == "dispatch":
         params = load_and_validate(args.input_file)
 
         handler = HANDLERS[JobType(params["benchmark_name"])](args, params, job_manager)
         handler.dispatch_handler()
+
     elif args.action == "poll":
         job_id = args.job_id
         job = job_manager.get_job(job_id)
         job_type = job["job_type"]
 
         handler = HANDLERS[JobType(job_type)](args, None, job_manager)
         handler.poll_handler()
+
+    elif args.action == "list-jobs":
+        list_jobs(args, job_manager)
+
     else:
-        logging.error("Invalid action specified. Use 'dispatch' or 'poll'.")
+        logging.error("Invalid action specified. Run with --help for usage information.")
         return 1
 
     return 0
 
 
+def list_jobs(args: argparse.Namespace, job_manager: JobManager) -> int:
+    """List jobs recorded in the job manager.
+
+    Args:
+        args: Parsed arguments.
+        job_manager: Job manager instance.
+    Returns:
+        Return code.
+    """
+    # Retrieve all jobs from JobManager.
+    jobs = job_manager.get_jobs()
+
+    # Apply filters if specified.
+    if args.filter and args.value:
+        jobs = [job for job in jobs if str(job.get(args.filter, "")).lower() == args.value.lower()]
+
+    # Display jobs in a tabular format.
+    if not jobs:
+        print("No jobs found.")
+        return 0
+
+    # Prepare data for tabulation.
+    headers = ["ID", "Backend", "Type", "Provider", "Qubits", "Shots"]
+    table = [
+        [
+            job.get("id", ""),
+            job.get("backend", ""),
+            job.get("job_type", ""),
+            job.get("provider", ""),
+            job.get("qubits", ""),
+            job.get("shots", ""),
```

Comment:
Things like qubits, and shots may be specific to a job type. Imo they should go into a misc column where we put metadata of the job.
Whereas id, backend, joy_type, provider should be there for every job, so they should have their dedicated columns. This is the same argument I was making for id, backend, provider not being in the schemas.

Reply:
Updated in 9bf8405. LMK if this is what you have in mind (tests were also updated here to support the changes in the `list-jobs` output).